/*
 * Copyright (c) 2010-2026. Axon Framework
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package messagingconcepts.componentmessageintercepting.entitydeclarative;

import org.axonframework.messaging.eventhandling.gateway.EventAppender;

public class GiftCard {

    private String cardId;
    private boolean redeemed;

    public void redeem(RedeemGiftCard command, EventAppender eventAppender) {
        eventAppender.append(new GiftCardRedeemed(command.cardId()));
    }

    void on(GiftCardRedeemed event) {
        this.cardId = event.cardId();
        this.redeemed = true;
    }

    public boolean isRedeemed() {
        return redeemed;
    }
}
